### DISCUSSION TITLE
Deployment migration & `build` folder tracking

### DISCUSSION THREAD

**Discussion by fermentfan** [ConsenSys-archive/truffle#5630] [General Q&A]
I have a problem understanding the development lifecycle including network deployments with Truffle. We're currently looking at how to do this in general and have built a repository based on Truffle + Ganache including Typescript support. I like the migration feature especially because we're deploying a proxy contract and need to override the linked implementation contract in the proxy after deploying it. Now I was looking at how the whole migration flow works and it seems like Truffle reads out the contract JSON inside the `build` folder to get the address of the currently deployed migrations contract.

Reading multiple threads on the internet and looking at the data inside the JSON (absolute directory paths??) it looks like it's not intended to track this folder via GIT. But how does the migration stack then know about the currently deployed contracts?

**edit:** I found the following dependency that solves this issue by extracts the deployment state tracking to another file: https://medium.com/medxprotocol/continuous-deployment-with-truffle-c4dfd4fcfd96

**Comment by lsqproduction** [comment]
Hello, @DennisVonDerBey. Thank you for opening this. 
We want to also recommend checking out `truffle networks` https://trufflesuite.com/docs/truffle/reference/truffle-commands/#networks as an alternative to the methods mentioned in the article above. 

As for the actual issue, we will discuss it with our team further for a way forward. stay tuned. 

@gnidan what do you think?

**Comment by gnidan** [comment]
This is tricky... people use git or npm for this purpose, but regarding those:
- I maintain git is the wrong tool for the job, since deployment tracking is not part of "source", and thus, strictly speaking, doesn't belong in "source control"
- NPM is centralized

Moving forward, I think maybe EthPM might be a good solution, but it uses IPFS and thus pinning becomes a concern. Truffle has considered making a centralized store for these assets ourselves, but that's, again, centralized.

Hopefully a path forward here will become clear in the near future.
